src/Integrations/Calculator/SettingsCalculator.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * Calculator Settings class.
+ *
+ * @package EightshiftForms\Integrations\Calculator
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Integrations\Calculator;
+
+use EightshiftFormsVendor\EightshiftFormsUtils\Helpers\UtilsSettingsHelper;
+use EightshiftFormsVendor\EightshiftFormsUtils\Settings\UtilsSettingGlobalInterface;
+use EightshiftFormsVendor\EightshiftFormsUtils\Helpers\UtilsSettingsOutputHelper;
+use EightshiftFormsVendor\EightshiftLibs\Services\ServiceInterface;
+
+/**
+ * SettingsCalculator class.
+ */
+class SettingsCalculator implements UtilsSettingGlobalInterface, ServiceInterface
+{
+	/**
+	 * Filter global settings key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_NAME = 'es_forms_settings_global_calculator';
+
+	/**
+	 * Filter settings global is Valid key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_IS_VALID_NAME = 'es_forms_settings_global_is_valid_calculator';
+
+	/**
+	 * Settings key.
+	 */
+	public const SETTINGS_TYPE_KEY = 'calculator';
+
+	/**
+	 * Calculator Use key.
+	 */
+	public const SETTINGS_CALCULATOR_USE_KEY = 'calculator-use';
+
+	/**
+	 * Register all the hooks
+	 *
+	 * @return void
+	 */
+	public function register(): void
+	{
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_NAME, [$this, 'getSettingsGlobalData']);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_IS_VALID_NAME, [$this, 'isSettingsGlobalValid']);
+	}
+
+	/**
+	 * Determine if settings global are valid.
+	 *
+	 * @return boolean
+	 */
+	public function isSettingsGlobalValid(): bool
+	{
+		$isUsed = UtilsSettingsHelper::isOptionCheckboxChecked(self::SETTINGS_CALCULATOR_USE_KEY, self::SETTINGS_CALCULATOR_USE_KEY);
+
+		if (!$isUsed) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get global settings array for building settings page.
+	 *
+	 * @return array<int, array<string, mixed>>
+	 */
+	public function getSettingsGlobalData(): array
+	{
+		// Bailout if feature is not active.
+		if (!UtilsSettingsHelper::isOptionCheckboxChecked(self::SETTINGS_CALCULATOR_USE_KEY, self::SETTINGS_CALCULATOR_USE_KEY)) {
```

Comment:
```suggestion
		if (!$this->isSettingsGlobalValid()) {
```

As this is extracted above, we can use it.